src/libraries/System.Text.Json/src/System/Text/Json/JsonHelpers.Escaping.cs

```diff
@@ -53,6 +53,43 @@ public static byte[] GetEscapedPropertyNameSection(ReadOnlySpan<byte> utf8Value,
             return escapedString;
         }
 
+        public static string GetPropertyName(
```

Comment:
Unlike the source generator, escaping shouldn't apply to runtime components. This is already handled by `JsonSerializerOptions.Encoder` property which by default escapes characters to their unicode representation:

```C#
Console.WriteLine(JsonSerializer.Serialize(new Foo { Name = "test" })); // {"\u0022\u0027":"test"}

public class Foo
{
    [JsonPropertyName("\"'")]
    public string Name { get; set; }
}
```

Any fix to JsonNode.Path should try to be consistent with what the serializer is doing already, i.e. encode single-quote characters to \u0027.

Reply:
@eiriktsarpalis , fixed by https://github.com/dotnet/runtime/pull/84688/commits/338a084cdde4008fc4d7d32eb370cd0fd5351501